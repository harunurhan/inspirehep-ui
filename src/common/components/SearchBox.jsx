import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Input } from 'antd';

class SearchBox extends Component {
  render() {
    return (
      <Input.Search
        style={{ verticalAlign: 'middle' }}
        addonBefore={this.props.searchScopeName}
        placeholder={this.props.placeholder}
        defaultValue={this.props.defaultValue}
        onSearch={this.props.onSearch}
        enterButton
      />
    );
  }
}

SearchBox.propTypes = {
  defaultValue: PropTypes.string,
  searchScopeName: PropTypes.string,
  placeholder: PropTypes.string,
  onSearch: PropTypes.func,
};

SearchBox.defaultProps = {
  defaultValue: null,
  placeholder: null,
  onSearch: null,
  searchScopeName: null,
};

export default SearchBox;
