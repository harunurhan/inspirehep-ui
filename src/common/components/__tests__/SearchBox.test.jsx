import React from 'react';
import { shallow } from 'enzyme';

import SearchBox from '../SearchBox';

describe('SearchBox', () => {
  it('render initial state with all props set', () => {
    const wrapper = shallow(
      <SearchBox
        value="value"
        placeholder="placeholder"
        searchScopeName="scope"
        onSearch={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders new value on change', () => {
    const wrapper = shallow(<SearchBox value="value" />);
    wrapper.instance().onChange({ target: { value: 'new' } });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
