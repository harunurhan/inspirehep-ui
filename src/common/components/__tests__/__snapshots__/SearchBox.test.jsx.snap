// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox render initial state with all props set 1`] = `
<Search
  addonBefore={<Connect(SearchScopeSelectContainer) />}
  enterButton={true}
  inputPrefixCls="ant-input"
  onChange={[Function]}
  onSearch={[Function]}
  placeholder="placeholder"
  prefixCls="ant-input-search"
  size="large"
  style={
    Object {
      "verticalAlign": "middle",
    }
  }
  value="value"
/>
`;

exports[`SearchBox renders new value on change 1`] = `
<Search
  addonBefore={<Connect(SearchScopeSelectContainer) />}
  enterButton={true}
  inputPrefixCls="ant-input"
  onChange={[Function]}
  onSearch={null}
  placeholder={null}
  prefixCls="ant-input-search"
  size="large"
  style={
    Object {
      "verticalAlign": "middle",
    }
  }
  value="new"
/>
`;
